File: Golang/simple_factory/pizza_store.go
package simple_factory

import "fmt"


type pizzaStore struct {
	factory pizzaFactory
}


func NewPizzaStore(factory pizzaFactory) *pizzaStore {
	return &pizzaStore{
		factory: factory,
	}
}


func (ps *pizzaStore) OrderPizza(pizzaType string) {

	if pizza, err := ps.factory.CreatePizza(pizzaType); err != nil {
		fmt.Println(err.Error())
	} else {
		pizza.Prepare()
		pizza.Bake()
		pizza.Cut()
		pizza.Box()
	}
}

File: Python/simple_factory/pizza_store.py
from pizza import Pizza
from pizza_factory import SimplePizzaFactory


class PizzaStore:
    factory: SimplePizzaFactory

    def __init__(self, factory: SimplePizzaFactory):
        self.factory = factory

    def order_pizza(self, type_: str) -> Pizza:
        pizza: Pizza

        pizza = self.factory.create_pizza(type_)

        pizza.prepare()
        pizza.bake()
        pizza.cut()
        pizza.box()

        return pizza
